kivy/core/image/__init__.py

```diff
@@ -644,10 +685,35 @@ def _set_anim_delay(self, x):
                 self._anim_ev = None
 
             if self._anim_delay >= 0:
-                self._anim_ev = Clock.schedule_interval(self._anim,
+                if not self.durations:
+                    self._anim_ev = Clock.schedule_interval(self._anim,
+                                                            self._anim_delay)
+                else:
+                    self._anim_ev = Clock.schedule_once(self._anim,
                                                         self._anim_delay)
 
     anim_delay = property(_get_anim_delay, _set_anim_delay)
+
+    def _get_durations(self):
+        return self._durations
+
+    def _set_durations(self, x):
+        # Waiting for list of ints or floats in microseconds like in GIF format, not seconds
+        if x is None:
+            self._durations = None
+            return
+        if not(isinstance(x, list) and all(isinstance(elem, (int, float)) for elem in x)) and len(x) > 0:
+            raise Exception('Property durations must be list of int or float with len > 0')
+
+        if all(elem == x[0] for elem in x[1:]):
+            # durations are constant, so we have standart anim_delay case
+            self._durations = None
+        else:
+            self._durations = [elem / 1000 for elem in x]
```

Comment:
i think we shouldn't change the values here, if i put a value x in a property, i expect to find x in the property later, not x / 1000.